portal/src/it/scala/cromwell/pipeline/controller/ProjectControllerItTest.scala

```diff
@@ -0,0 +1,169 @@
+package cromwell.pipeline.controller
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.ContentTypes.`application/json`
+import akka.http.scaladsl.model.{HttpEntity, StatusCodes}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import com.dimafeng.testcontainers.{ForAllTestContainer, PostgreSQLContainer}
+import com.typesafe.config.Config
+import cromwell.pipeline.ApplicationComponents
+import cromwell.pipeline.controller.AuthController._
+import cromwell.pipeline.datastorage.dto.{Project, ProjectAdditionRequest, ProjectId, User}
+import cromwell.pipeline.utils.auth.{AccessTokenContent, TestContainersUtils, TestUserUtils}
+import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport._
+import org.scalatest.compatible.Assertion
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.time.{Millis, Seconds, Span}
+import org.scalatest.{AsyncWordSpec, Matchers}
+import org.scalatestplus.mockito.MockitoSugar
+import play.api.libs.json.Json
+
+class ProjectControllerItTest
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ScalatestRouteTest
+    with ForAllTestContainer {
+
+  override val container: PostgreSQLContainer = TestContainersUtils.getPostgreSQLContainer()
+  container.start()
+  implicit val config: Config = TestContainersUtils.getConfigForPgContainer(container)
+  private val components: ApplicationComponents = new ApplicationComponents()
+
+  implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(timeout = Span(5, Seconds), interval = Span(500, Millis))
+
+  override protected def beforeAll(): Unit =
+    components.datastorageModule.pipelineDatabaseEngine.updateSchema()
+
+  import components.controllerModule.projectController
+  import components.datastorageModule.{projectRepository, userRepository}
+
+
+  private val userPassword = "-Pa$$w0rd-"
+
+  "ProjetController" when {
+
+    "GET project by id" should {
+      "return project entity" in {
+        val dummyUser: User = TestUserUtils.getDummyUser(password = userPassword)
+        val userId = dummyUser.userId
+        val dummyProject = TestUserUtils.getDummyProject(userId)
+        val projectId = dummyProject.projectId
+        val accessToken = AccessTokenContent(userId.value)
+
+        whenReady(userRepository.addUser(dummyUser)) { _ =>
```

Comment:
Why do you need `whenReady`?
It could be done `Future.andThen`, right?

Reply:
fixed